pyo3-macros-backend/src/intopydict.rs

```diff
@@ -0,0 +1,366 @@
+            if attr_type.starts_with(collection) {
+                let attr_type = attr_type.replace('>', "");
+                let attr_list: Vec<&str> = attr_type.split('<').collect();
+                let out = Self::handle_collection(&attr_list, span);
+
+                return out.unwrap();
+            }
+        }
+
+        match attr_type {
+            "i8" | "i16" | "i32" | "i64" | "i128" | "isize" | "u8" | "u16" | "u32" | "u64"
+            | "u128" | "usize" | "f32" | "f64" | "char" | "bool" | "&str" | "String" => {
+                Pyo3Type::Primitive
+            }
+            _ => Pyo3Type::NonPrimitive,
+        }
+    }
+
+    fn handle_collection(attr_type: &[&str], span: Span) -> syn::Result<Pyo3Type> {
+        match attr_type[0] {
+            "BTreeSet" | "BinaryHeap" | "Vec" | "HashSet" | "LinkedList" | "VecDeque" => {
+                Ok(Pyo3Type::CollectionSing(Box::new(
+                    Self::handle_collection(&attr_type[1..], span).unwrap(),
+                )))
+            }
+            "BTreeMap" | "HashMap" => {
+                Err(Error::new(span, "Derive currently doesn't support map types. Please use a custom implementation for structs using a map type like HashMap or BTreeMap"))
+            }
+            "i8" | "i16" | "i32" | "i64" | "i128" | "isize" | "u8" | "u16" | "u32" | "u64"
+            | "u128" | "usize" | "f32" | "f64" | "char" | "bool" | "&str" | "String" => {
+                Ok(Pyo3Type::Primitive)
+            }
+            _ => Ok(Pyo3Type::NonPrimitive),
+        }
+    }
+}
+
+impl Parse for Pyo3Collection {
+    fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
+        let tok_stream: TokenStream = input.parse()?;
+        let mut binding = tok_stream
+            .to_string()
+            .as_str()
+            .replace(|c| c == ' ' || c == '{' || c == '}' || c == '\n', "");
+
+        if !binding.contains(':') {
+            return Ok(Pyo3Collection(Vec::new()));
+        }
+
+        if binding.as_bytes()[binding.len() - 1] as char != ',' {
+            binding.push(',');
+        }
+
+        let name_map = split_struct(binding);
+
+        let mut field_collection: Vec<Pyo3DictField> = Vec::new();
+
+        for (field_name, (field_val, dict_key)) in &name_map {
+            field_collection.push(Pyo3DictField::new(
+                field_name.to_string(),
+                field_val,
+                input.span(),
+                dict_key.clone(),
+            ))
+        }
+
+        Ok(Pyo3Collection(field_collection))
+    }
+}
+
+fn split_struct(binding: String) -> HashMap<String, (String, Option<String>)> {
+    let mut stack: Vec<char> = Vec::new();
+    let mut start = 0;
+    let binding = binding.replace('\n', "");
+    let mut name_map: HashMap<String, (String, Option<String>)> = HashMap::new();
+
+    for (i, char_val) in binding.chars().enumerate() {
+        if char_val == ',' && stack.is_empty() {
+            if binding[start..i].starts_with('#') {
```

Comment:
I think you might want to take a look at `take_pyo3_options` to see how the rest of the crate parses this.